src/abstractblockarray.jl

```diff
@@ -180,6 +180,10 @@ end
 @inline Base.getindex(A::AbstractMatrix, kr::AbstractVector, jr::Block) = ArrayLayouts.layout_getindex(A, kr, jr)
 @inline Base.getindex(A::AbstractMatrix, kr::BlockRange{1}, jr::BlockRange{1}) = ArrayLayouts.layout_getindex(A, kr, jr)
 @inline Base.getindex(A::LayoutMatrix, kr::BlockRange{1}, jr::BlockRange{1}) = ArrayLayouts.layout_getindex(A, kr, jr)
+# Treat the case of linear indexing of blocks
+@inline Base.getindex(A::AbstractVector{T}, kr::Block{1}) where {T} = ArrayLayouts.layout_getindex(A, kr)
+@inline Base.getindex(A::AbstractArray{T,N}, kr::Block{1}) where {T,N} =
```

Comment:
@putianyi889 I thought you already added support for linear indexing? Could you look at this and see why he needed to re-add it?

Reply:
It's because I added linear indexing for `AbstractBlockArray`, not for `AbstractArray`.

Reply:
Currently block indexing for concrete arrays is kinda weird.
```julia
julia> A=rand(10,10);

julia> A[Block(1)]
10-element Array{Float64,1}:
 0.018449731365326905
 0.43500654038664743
 0.2877990218367896
 0.2683321023115157
 0.09401161807946257
 0.7472157420605761
 0.3997084382794658
 0.14882860945611087
 0.4323069792050045
 0.4532879831382344
```

Reply:
Right, that's the bug. it should just return `A`

Reply:
Yes, and that's what I added a few lines of code for in this pull request, which seemed all that was needed. A linear index of a Block would be converted to a multidimensional index (using ind2sub with the blocksize of A).

Reply:
Indeed for AbstractBlockArray it was already working

Reply:
I think you should modify the backend which is `view`.
```julia
julia> @which view(A,Block(1)) #bugged
view(A::AbstractArray, I::Vararg{Any,N}) where N in Base at subarray.jl:155

julia> @which view(A,Block(1,1)) #correct
view(A::AbstractArray, I::Vararg{Any,N}) where N in Base at subarray.jl:155
```

Reply:
With your current changes, `A[Block(1)]` works but `view(A,Block(1))` is still not working.

Reply:
Could you suggest where? Or feel free to implement, we can also discard this pull request here. Indeed I did not test for that.

Reply:
Can I create a PR inside a PR? I have never heard this trick before.